src/Benchmarks/FortunesDapperMiddleware.cs

```diff
@@ -46,14 +46,13 @@ public FortunesDapperMiddleware(RequestDelegate next, string connectionString, D
 
         private static async Task<IEnumerable<Fortune>> LoadRows(string connectionString, DbProviderFactory dbProviderFactory)
         {
-            var result = new List<Fortune>();
+            List<Fortune> result;
 
             using (var db = dbProviderFactory.CreateConnection())
             {
                 db.ConnectionString = connectionString;
-                await db.OpenAsync();
-
-                result.AddRange(await db.QueryAsync<Fortune>("SELECT [Id], [Message] FROM [Fortune]"));
+                // note: don't need to open connection if only doing one thing; let dapper do it
+                result = (await db.QueryAsync<Fortune>("SELECT [Id], [Message] FROM [Fortune]")).AsList();
```

Comment:
If you mean "to match the expected data": then no, dapper doesn't do that, because it is consuming a raw `IDataReader` (`DbDataReader` on core-clr) API, and **doesn't know** the number of rows until it reaches the end. In this setup, it is essentially:

    List<T> buffer = new List<T>();
    while (await reader.ReadAsync(cancel).ConfigureAwait(false)) {
        buffer.Add(func(reader));
    }
    while (await reader.NextResultAsync(cancel).ConfigureAwait(false)) { }

(where `func` is a delegate that processes each row into the target data type)